### PR TITLE
[24452] Add Topic-Name Profile Lookup for Entity Creation 

### PR DESCRIPTION
**Pull request by ZakariaTalbi**
Included feature for topic profile lookup with tests to corroborate the correct implementation and behaviour of the feature.

### CODE REVIEW

File: docs/rst/notes/forthcoming_version.rst [hunk 1]
```diff
@@ -5,3 +5,7 @@
 ###################
 Forthcoming Version
 ###################
+
+This release includes the following **features**:
+
+- Support per-topic DataReader QoS customization via XML profiles in ``init_monitor``.
```
Comment (zesk1999) [inline docs/rst/notes/forthcoming_version.rst]: By convention, we leave a new line in the end of the files.
```suggestion
- Support per-topic DataReader QoS customization via XML profiles in ``init_monitor``.

```

File: test/unittest/StatisticsBackend/profiles/datareader_profile.xml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"  ?>
+<profiles>
+    <data_reader profile_name="_fastdds_statistics_history2history_latency">
+        <qos>
+            <durability>
+                <kind>VOLATILE</kind>
+            </durability>
+        </qos>
+    </data_reader>
+</profiles>
```
Comment (zesk1999) [inline test/unittest/StatisticsBackend/profiles/datareader_profile.xml:10]:
```suggestion
</profiles>

```

File: src/cpp/StatisticsBackend.cpp [hunk 3]
```diff
@@ -338,7 +338,16 @@ EntityId create_and_register_monitor(
         }
 
         /* Create DataReaders */
-        if (topic == MONITOR_SERVICE_TOPIC)
+        fastdds::dds::DataReaderQos unused_qos;
+        if (monitor->subscriber->get_datareader_qos_from_profile(topic, unused_qos) == RETCODE_OK)
```
Comment (zesk1999) [inline src/cpp/StatisticsBackend.cpp]: The name unused_qos is a bit contraintuitive, I would consider renaming it so it does not seem like a redundant variable.
```suggestion
        fastdds::dds::DataReaderQos profile_check_qos;
        if (monitor->subscriber->get_datareader_qos_from_profile(topic, profile_check_qos) == RETCODE_OK)
```

File: src/cpp/StatisticsBackend.cpp [hunk 4]
```diff
@@ -338,7 +338,16 @@ EntityId create_and_register_monitor(
...
+        {
+            monitor->statistics_readers[topic] = monitor->subscriber->create_datareader_with_profile(
+                monitor->statistics_topics[topic],
+                topic,
+                monitor->statistics_reader_listener,
+                StatusMask::all());
+        }
+        else if (topic == MONITOR_SERVICE_TOPIC)
```
Comment (zesk1999) [inline src/cpp/StatisticsBackend.cpp]: I would put this condition first because, in the edge case of someone tuning this topic through an xml profile where only some QoS fields are specified, max_instances, max_samples_per_instance and max_samples would get the default -1 (unlimited), which is not what we want for a monitor service reader.

Comment (zesk1999) [reply src/cpp/StatisticsBackend.cpp]: In case that you agree on processing this corner case, I would suggest adding a regression test with a monitor service override xml profile where we would set one QoS field that is neither of above mentioned max fields and then checking if max_instances=1500 and max_samples=1600.

**Review by zesk1999 — CHANGES_REQUESTED** [review]
Great job!
I just have a few nits that cover some specific cases and give the consistency over the repository.
